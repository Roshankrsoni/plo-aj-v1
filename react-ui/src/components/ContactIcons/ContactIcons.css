.horizontally-placed{
    display:flex;
    margin-top:0;
}
.horizontally-placed .subsection-data + .subsection-data{
    padding-left:3rem;
    
}

.darkmode .contactimg{
    filter: invert(100%) sepia(5%) saturate(320%) hue-rotate(241deg) brightness(120%) contrast(90%);
}
.contactimg{
    width: 2rem;
    height:2rem;
    transition: all 0.5s ease-in-out;
}

.contactimg:hover{
    filter: invert(51%) sepia(21%) saturate(5271%) hue-rotate(328deg) brightness(99%) contrast(86%);
    transform: scale(1.2);
}


.rightpane{
    width: 3.5rem;
    flex-wrap: wrap;
    position: fixed;
    background-color: rgba(219, 219, 219, 0.5);
    padding:1rem;
    right:0;
    top:50%;
    transform: translateY(-50%);
    border-radius: 0.25rem 0 0  0.25rem;
    animation: slideIn 1s ease-in-out forwards;
    z-index: 999999;
}
.darkmode .rightpane{
    background-color: rgba(219, 219, 219, 0.2);
}
.rightpane.horizontally-placed .subsection-data{
    margin-top:0;
}

.rightpane.horizontally-placed .subsection-data + .subsection-data{
    padding-left:0;
    margin-top:1.5rem;
}

.rightpane .contactimg{
    width: 1.5rem;
    height:1.5rem;
}

@media (max-width: 1200px) { 
    .rightpane{
        top:45vH;
        width: 4rem;
    }
    .rightpane .contactimg{
        width: 2rem;
        height:2rem;
    }
    .rightpane.horizontally-placed .subsection-data + .subsection-data{
        margin-top:2rem;
    }

}

@keyframes slideIn{
    0%{
        right:-2.5rem;
    }
    80%{
        right:0.5rem;
    }
    100%{
        right:0;
    }
}