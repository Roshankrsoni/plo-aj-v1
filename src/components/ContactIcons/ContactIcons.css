.horizontally-placed {
  display: flex;
  margin-top: 0;
}
.horizontally-placed .subsection-data + .subsection-data {
  padding-left: 3rem;
}

.darkmode .contactimg {
  filter: invert(100%) sepia(5%) saturate(320%) hue-rotate(241deg)
    brightness(120%) contrast(90%);
}
.contactimg {
  width: 2rem;
  height: 2rem;
  transition: all 0.3s ease-in-out;
}

.contactimg:hover {
  filter: invert(51%) sepia(21%) saturate(5271%) hue-rotate(328deg)
    brightness(99%) contrast(86%);
  transform: scale(1.2);
}


.inside-leftpane {
  margin-top: 2.5rem;
  padding: 1rem 2rem 0 2rem;
  position: relative;
  left: -2rem;
  width: calc(100% + 4rem);
  border-top: 1px solid var(--grey5);
}

.inside-leftpane.horizontally-placed .subsection-data {
  margin-top: 0;
}

.inside-leftpane.horizontally-placed .subsection-data + .subsection-data {
  padding-left: 1.3rem;
}

.inside-leftpane .contactimg {
  width: 1.5rem;
  height: 1.5rem;
}

@media (max-width: 1200px) {
  .rightpane {
    top: 45vh;
    width: 4rem;
  }
  .rightpane .contactimg {
    width: 2rem;
    height: 2rem;
  }
  .rightpane.horizontally-placed .subsection-data + .subsection-data {
    margin-top: 2rem;
  }
}

@keyframes slideIn {
  0% {
    right: -2.5rem;
  }
  80% {
    right: 0.5rem;
  }
  100% {
    right: 0;
  }
}
