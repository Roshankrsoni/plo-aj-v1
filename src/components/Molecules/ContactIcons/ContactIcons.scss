.horizontally-placed {
  display: flex;
  margin-top: 0;
}
.horizontally-placed .subsection-data + .subsection-data {
  padding-left: 3rem;
}

.darkmode .contactimg {
  filter: invert(100%) sepia(5%) saturate(320%) hue-rotate(241deg)
    brightness(120%) contrast(90%);
}
.contactimg {
  width: 1.5rem;
  height: 1.5rem;
  stroke: var(--grey1);
}

.contactimg:hover {
  filter: invert(51%) sepia(21%) saturate(5271%) hue-rotate(328deg)
    brightness(99%) contrast(86%);
}

.inside-leftpane {
  padding: 1rem 1.5rem;
  position: relative;
  border-top: 1px solid var(--grey5);

  width: 100%;
  display: flex;
  justify-content: space-between;
}

.inside-leftpane .subsection-data {
  margin-top: 0;
}

.inside-leftpane .subsection-data + .subsection-data {
  padding-left: 1rem;
}

.inside-leftpane .contactimg {
  width: 1.3rem;
  height: 1.3rem;
}

@media (max-width: 1200px) {
  .rightpane {
    top: 45vh;
    width: 4rem;
  }
  .rightpane .contactimg {
    width: 2rem;
    height: 2rem;
  }
  .rightpane.horizontally-placed .subsection-data + .subsection-data {
    margin-top: 2rem;
  }
}

@keyframes slideIn {
  0% {
    right: -2.5rem;
  }
  80% {
    right: 0.5rem;
  }
  100% {
    right: 0;
  }
}
