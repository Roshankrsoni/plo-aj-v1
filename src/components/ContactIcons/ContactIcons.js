import React from "react";
import "./ContactIcons.css";

import email from "../../assets/email.svg";
import linkedin from "../../assets/linkedin.svg";
import github from "../../assets/github.svg";
import insta from "../../assets/insta.svg";

const ContactIcons = (props) => {
  return (
    <div className={"horizontally-placed " + props.className}>
      <div className="subsection-data">
        <a
          href={"mailto: " + props.data.contactDetails.emailId}
          title="Email Address"
        >
          <img className="contactimg" src={email} alt="contact img"></img>
        </a>
      </div>
      <div className="subsection-data">
        <a
          href={props.data.contactDetails.linkedin}
          rel="noopener noreferrer"
          target="_blank"
          title="Linkedin"
        >
          <img className="contactimg" src={linkedin} alt="linkedin img"></img>
        </a>
      </div>
      <div className="subsection-data">
        <a
          href={
            "https://www.instagram.com/" + props.data.contactDetails.instaId
          }
          rel="noopener noreferrer"
          target="_blank"
          title="Instagram"
        >
          <img className="contactimg" src={insta} alt="Instagram img"></img>
        </a>
      </div>

      <div className="subsection-data">
        <a
          href={props.data.contactDetails.github}
          target="_blank"
          rel="noopener noreferrer"
          title="Github"
        >
          <img className="contactimg" src={github} alt="github img"></img>
        </a>
      </div>
    </div>
  );
};

export default ContactIcons;
