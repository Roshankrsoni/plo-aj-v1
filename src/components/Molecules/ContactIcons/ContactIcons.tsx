import React from "react";
import "./ContactIcons.scss";
import { useGlobalContext } from "../../../context/GlobalContext";
import { FiGithub, FiInstagram, FiLinkedin, FiMail } from "react-icons/fi";

const ContactIcons = ({ className }: { className: string }) => {
  const { profileData } = useGlobalContext();
  const { contactDetails } = profileData;
  const { emailId, linkedin, instaId, github } = contactDetails;
  return (
    <div className={"horizontally-placed " + className}>
      <div className="subsection-data">
        <a href={"mailto: " + emailId} title="Email Address">
          <FiMail className="contactimg" title="email address" />
        </a>
      </div>
      <div className="subsection-data">
        <a
          href={linkedin}
          rel="noopener noreferrer"
          target="_blank"
          title="Linkedin"
        >
          <FiLinkedin className="contactimg" title="linkedin link" />
        </a>
      </div>
      <div className="subsection-data">
        <a
          href={"https://www.instagram.com/" + instaId}
          rel="noopener noreferrer"
          target="_blank"
          title="Instagram"
        >
          <FiInstagram className="contactimg" title="instagram link" />
        </a>
      </div>

      <div className="subsection-data">
        <a
          data-testid="github"
          href={github}
          target="_blank"
          rel="noopener noreferrer"
          title="Github"
        >
          <FiGithub className="contactimg" title="github link" />
        </a>
      </div>
    </div>
  );
};

export default ContactIcons;
export { ContactIcons };
