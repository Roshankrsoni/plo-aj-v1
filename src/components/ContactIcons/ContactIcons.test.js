import React from 'react';
import { shallow } from 'enzyme';
import ContactIcons from './ContactIcons';

import dummyData from '../TestData/testData.json'

describe('MyComponent', () => {
    it('should render correctly in "debug" mode', () => {
        const component = shallow(<ContactIcons data={dummyData} />);
        expect(component).toMatchSnapshot();
    });
});
