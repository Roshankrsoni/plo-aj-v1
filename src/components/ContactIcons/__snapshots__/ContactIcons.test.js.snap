// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render correctly in "debug" mode 1`] = `
<div
  className="horizontally-placed undefined"
>
  <div
    className="subsection-data"
  >
    <a
      href="mailto: anoopjadhav@gmail.com"
      title="Email Address"
    >
      <img
        alt="contact img"
        className="contactimg"
        src="email.svg"
      />
    </a>
  </div>
  <div
    className="subsection-data"
  >
    <a
      href="https://www.linkedin.com/in/anoop-jadhav-44528258"
      rel="noopener noreferrer"
      target="_blank"
      title="Linkedin"
    >
      <img
        alt="linkedin img"
        className="contactimg"
        src="linkedin.svg"
      />
    </a>
  </div>
  <div
    className="subsection-data"
  >
    <a
      href="https://www.instagram.com/anoop.designs"
      rel="noopener noreferrer"
      target="_blank"
      title="Instagram"
    >
      <img
        alt="Instagram img"
        className="contactimg"
        src="insta.svg"
      />
    </a>
  </div>
  <div
    className="subsection-data"
  >
    <a
      href="https://github.com/Anoopjadhav"
      rel="noopener noreferrer"
      target="_blank"
      title="Github"
    >
      <img
        alt="github img"
        className="contactimg"
        src="github.svg"
      />
    </a>
  </div>
</div>
`;
